### PR TITLE
Provide new API to set corner radii. Some internal refactoring.

### PR DESCRIPTION
**Pull request by IlyaGulya**
Made as separate PR from #173

### CODE REVIEW

File: roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java [hunk 1]
```diff
@@ -1,17 +1,14 @@
 package com.makeramen.roundedimageview;
 
-import android.support.annotation.IntDef;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
+public enum Corner {
```
Comment (vinc3m1) [inline roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java]: this doesn't seem to be an improvement over the intdef?

File: roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
-})
-public @interface Corner {
-  int TOP_LEFT = 0;
-  int TOP_RIGHT = 1;
-  int BOTTOM_RIGHT = 2;
-  int BOTTOM_LEFT = 3;
+  public int index;
+
+  Corner(final int index) {
+    this.index = index;
```
Comment (vinc3m1) [inline roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java]: enum.ordinal() should be sufficient no?

File: roundedimageview/src/main/java/com/makeramen/roundedimageview/Corners.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  TOP(EnumSet.of(Corner.TOP_LEFT, Corner.TOP_RIGHT)),
+  BOTTOM(EnumSet.of(Corner.BOTTOM_LEFT, Corner.BOTTOM_RIGHT)),
+  LEFT(EnumSet.of(Corner.TOP_LEFT, Corner.BOTTOM_LEFT)),
+  RIGHT(EnumSet.of(Corner.TOP_RIGHT, Corner.BOTTOM_RIGHT)),
+  OTHER_TOP_LEFT(EnumSet.of(Corner.TOP_RIGHT, Corner.BOTTOM_LEFT, Corner.BOTTOM_RIGHT)),
+  OTHER_TOP_RIGHT(EnumSet.of(Corner.TOP_LEFT, Corner.BOTTOM_LEFT, Corner.BOTTOM_RIGHT)),
+  OTHER_BOTTOM_LEFT(EnumSet.of(Corner.TOP_LEFT, Corner.TOP_RIGHT, Corner.BOTTOM_RIGHT)),
+  OTHER_BOTTOM_RIGHT(EnumSet.of(Corner.TOP_LEFT, Corner.TOP_RIGHT, Corner.BOTTOM_LEFT)),
+  DIAGONAL_FROM_TOP_LEFT(EnumSet.of(Corner.TOP_LEFT, Corner.BOTTOM_RIGHT)),
+  DIAGONAL_FROM_TOP_RIGHT(EnumSet.of(Corner.TOP_RIGHT, Corner.BOTTOM_LEFT));
```
Comment (vinc3m1) [inline roundedimageview/src/main/java/com/makeramen/roundedimageview/Corners.java]: i'm not sure why we need all these enum values. `Corners` should always have all corners with default radius of 0. Then we don't need the other `Corner` enum either.

Just need something like this right?
```
public class Corners() {
    float topLeft;
    float topRight;
    float bottomLeft;
    float bottomRight;
}
```

and maybe a helper `setAll(float radius)` function but that should be all

File: roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java (hunk 2 above)
Comment (IlyaGulya) [inline roundedimageview/src/main/java/com/makeramen/roundedimageview/Corner.java]: +

**Review by dev-aniketj — APPROVED** [review]

**Review by dev-aniketj — APPROVED** [review]

**Pull request closed without merging**